// splunk-sdk/index.js

// Copyright 2011 Splunk, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"): you may
// not use this file except in compliance with the License. You may obtain
// a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations
// under the License.

(function() {
    var root = exports || this;

    // Declare a process environment so that we can set
    // some globals here and have interop with node
    process.env = process.env || {};

    module.exports = root = {
        Logger          : require('splunk-sdk/lib/log').Logger,
        Context         : require('splunk-sdk/lib/context'),
        Service         : require('splunk-sdk/lib/service'),
        Http            : require('splunk-sdk/lib/http'),
        Utils           : require('splunk-sdk/lib/utils'),
        Async           : require('splunk-sdk/lib/async'),
        Paths           : require('splunk-sdk/lib/paths').Paths,
        Class           : require('splunk-sdk/lib/jquery.class').Class,
        ModularInputs   : require('splunk-sdk/lib/modularinputs')
    };
    
    if (typeof(window) === 'undefined') {
        root.NodeHttp = require('splunk-sdk/lib/platform/node/node_http').NodeHttp;
    }
})();
